src/utils/generate-html-id.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2018 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const _ = require('lodash');
+
+module.exports = {
+
+  /**
+   * Generates a valid HTML identifier from the string
+   * @param {string} text The text to use to generate the identifier
+   * @returns {string} the string for the identifier
+   */
+  generateId: text => _.kebabCase(text),
```

Comment:
For me, there are 2 reasons to do this:

1. Abstract the `id` generation logic which helps refactoring. When I rewrote this 3 times in the PR, I only updated this single method, not the rest of the code
2. Expose the method directly so I can have dedicated unit tests to test edge cases without having to run the whole pipeline. I test default use cases with the pipeline, and edge cases directly on the method (tests are faster, and maybe some edge cases are not achievable through the normal flow)

Reply:
Well, if you must leave it ;)

1. Normally I would wholeheartedly agree; it is a sign of a very good architecture to abstract things away so they can be reused! There is such a thing as overarchitecting though; each file and function and line of code we add adds a bit of overhead. Here we are adding an entire file for a single function that is called a single time and thename is a bit more nebulous than "kebabCase" and it's documentation doesn't really add much either; the intention of describing what the function is for is great btw, but the documentation doesn't really elaborate on what makes this function particularly great for that. I wholeheartedly agree that we should introduce abstraction, but maybe we start doing that when the function is used more than once or when it becomes more than an alias ;)
2. Yes and I think your tests are great. But so are lodash's and lodashs actually are a bit more elaborate and specific and they have been reviewed and used by a lot of people; they do contain tests specific to handling unicode (e.g. in test/words.js). I don't think we can approach that level of diligence or gain a lot by replicating that work – the chance of a bad regression in lodash that we can find with our tests is IMHO slim.

But I have spent far too much time elaborating on such a minor point and distracting from more important matters; if you decide to keep it that way, so be it! :)

Reply:
I'm actually good either way. I initially abstracted because the implementation was more complex and wanted the code to be re-usable (i.e. generate identifiers for other things than just headings), but as you pointed out, in the end with your improvements I'm just creating a proxy that doesn't add better tests nor documentation :)